hashintel/hash
# H-783: Add entity type icons

## 🌟 What is the purpose of this PR?




This PR adds editable entity type icons, and displays them in various places in the application.

## 🔗 Related links





- H-783

## 🔍 What does this change?




- adds a `EditEmojiIconButton` component, used to set emoji icons for pages and entity types
- fixes a bug where earlier versions of archived types were being displayed in the sidebar
- fixes a bug where all versions of a type were being displayed in the edit pinned entity type modal

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



## 🐾 Next steps



- display inherited entity type icon when entity type does not have an icon set (H-1367)
- display entity type icons in the types and entities data grid tables (H-1368)

## 🛡 What tests cover this?



None.

## ❓ How to test this?



⚠️ This PR cannot be tested in the deployment, as it requires backend changes.

1. Run HASH locally
2. Create an entity type, and edit its icon in the entity type editor and publish a new version of the type
3. Ensure the type's icon is viewable throughout the application

## 📹 Demo



<details>
  <summary>Entity Type icon in the editor and sidebar</summary>
  <img width="1190" alt="image" src="https://github.com/hashintel/hash/assets/42802102/f36ac408-5656-43ca-ad7b-2d3820ddd620">
</details>

<details>
  <summary>Entity Type icons in the entity type selector</summary>
<img width="496" alt="image" src="https://github.com/hashintel/hash/assets/42802102/b60f4836-0388-4ca7-b730-c4f3502d835c">
</details>

<details>
  <summary>Entity Type icons in the pinned entity type tab on the profile page</summary>
<img width="922" alt="image" src="https://github.com/hashintel/hash/assets/42802102/bcce54df-e0a6-4941-881b-a79973eb4ab6">
</details>

<details>
  <summary>Entity Type icons in the pinned entity type selector</summary>
<img width="547" alt="image" src="https://github.com/hashintel/hash/assets/42802102/2494caca-ecbe-4d59-a869-7aaee5ae76d3">
</details>

<details>
  <summary>Entity Type icons in mention suggester</summary>
<img width="359" alt="image" src="https://github.com/hashintel/hash/assets/42802102/8a74aaf9-2e1a-4dff-b5f8-101ed0e7236d">
</details>

<details>
  <summary>Entity Type icons in the an entity mention on a page</summary>
<img width="956" alt="image" src="https://github.com/hashintel/hash/assets/42802102/a92d3c53-003b-483d-9ffe-12caa60e1c85">
</details>
